### DISCUSSION TITLE
Tb-profiler fasta_profile feature

### DISCUSSION THREAD

**Discussion by Ankitapal1995** [jodyphelan/TBProfiler#179] [Q&A] [answered]
Hello, I have been using Tb-profiler from last 1 year for my research work. I am analysing AMR properties in M.tb using whole genome sequences in FASTA format. Till now I was using "fasta_profile" parameter to do all of my analysis but recently this features is replaced with "vcf_profile", now I am in big trouble as I am not able to use those fasta sequences for my analysis. If it is possible could you please give me your previous "fasta_profile" analysis code so that I can continue my analysis. Last but not the least, thank you so much for making such a great tool.

**Comment by jodyphelan** [comment, marked as answer]
Hi @Ankitapal1995 

The fasta_profile function should still be available. Does it give an error if you try to run it?

**Reply by Ankitapal1995** [reply]
Thanks for replying. Yes, It is giving an error while using FASTA sequence.

**Reply by jodyphelan** [reply]
Ok, would you mind sending the command you are running as well as the output error?
I assume you are using the latest verison?

**Reply by Ankitapal1995** [reply]
I am using TBProfiler version 2.3.0

**Reply by Ankitapal1995** [reply]
when i am using fasta_profile command:
tb-profiler fasta_profile 1733.5787.fna --dir test --csv

output error:
usage: tb-profiler [-h] {profile,vcf_profile,reprofile,reformat,load_library,update_tbdb,lineage,collate,version} ...
tb-profiler: error: invalid choice: 'fasta_profile' (choose from 'profile', 'vcf_profile', 'reprofile', 'reformat', 'load_library', 'update_tbdb', 'lineage', 'collate', 'version')
![tb-profiler_error](https://user-images.githubusercontent.com/61908837/135026607-33afb350-f407-40d4-aa53-ba1bc347a721.png)

**Reply by jodyphelan** [reply]
I think this might be an older version of tb-profiler which has installed (I also can't see the `create_db` function which should be present). Sometimes conda will downgrade an installation to a previous version when you install something new due to clashes with dependancies. You can check the version with `tb-profiler --version`. If it is not the most recent I would recommend forcing the latest version with `conda install tb-profiler=3.0.8`.

**Reply by Ankitapal1995** [reply]
I installed 3.08 version. Now It has fasta_profile parameter and working properly. Thank you so much for solving my issue.

**Reply by jodyphelan** [reply]
Great no problem!

**Comment by Ankitapal1995** [comment]
![issue_tb-profiler](https://user-images.githubusercontent.com/61908837/143562141-ba3cb23d-b1bb-40f5-acfd-635bb69338a7.png)

/anaconda3/lib/python3.9/site-packages/tbprofiler/text.py script has an error in line number 119 of tb-profiler lastest version 4.0.3, just because of this, it is throwing an error while processing the result in csv format. 
Please replace this below line in your script (line number : 119) 

>` text_strings["med_dp"] = json_results["qc"]["median_coverage"]`

with following line (by adding a comma between two keys)

> `text_strings["med_dp"] = json_results["qc"],["median_coverage"]`

Regards,
